### ISSUE TITLE
[suggestion] trash the entire backend, rewrite it in go or something else like it

### ISSUE THREAD

**Issue by skyhighwings** [PonyChat/Shuo#54]
It's been shown time and time again that the backend is full of bugs and other fun issues. #53, #51, #47, #45, amongst others (I didn't want to look at the entire list) show that this problem is deep-seated and will probably require a full rewrite in the end anyways. I'll look at the backend protocol sooner rather than later, and see what we can do on this matter.

**Comment by floogulinc** [comment]
The frontend would need to be rewritten as well (probably with Polymer), as right now its pretty entwined with the backend node app. If we make a backend that uses a standard JSON API plus something for the real time communication, we can build clients for the web, Android, and other platforms!

My experience is in Java and JS, so I would be able to work on it more if we stayed with node, but I understand why others may not like it.

I have been considering a full rewrite for a bit now, we may want to either make it a new project, or something like "Shuo 2.0" and keep up the current branch while the new one is in development.

**Comment by Xe** [comment]
I agree that some kind of protocol definition is needed. I think a good first step will be to define that kind of protocol.
